changelog_generator/generator.py

```diff
@@ -0,0 +1,115 @@
+import os
+from typing import List, NamedTuple, Optional, Sequence
+
+from jinja2 import Environment, FileSystemLoader
+
+from .commit import Commit
+from .repository_manager import RepositoryManager
+
+
+class CommitTree(NamedTuple):
+    commit_type: str
+    commits: Sequence[Commit]
+
+
+def render_changelog(
+    organization: str,
+    repository: str,
+    previous_tag: str,
+    current_tag: str,
+    commit_trees: Sequence[CommitTree],
+) -> str:
+    template_loader = FileSystemLoader(searchpath=os.path.dirname(__file__))
+    template_environment = Environment(loader=template_loader)
+    template = template_environment.get_template("changelog_template.jinja")
+
+    return template.render(
+        organization=organization,
+        repository=repository,
+        previous_tag=previous_tag,
+        current_tag=current_tag,
+        commit_trees=commit_trees,
+    )
+
+
+def get_commit_from_type(
+    commits: Sequence[Commit], commit_type: str
+) -> Sequence[Commit]:
+    return sorted(
+        filter(lambda commit: commit.commit_type == commit_type, commits),
+        key=lambda commit: commit.scope,
+    )
+
+
+def get_commit_but_types(
+    commits: Sequence[Commit], commit_types: Sequence[str]
+) -> Sequence[Commit]:
+    return sorted(
+        filter(
+            lambda commit: commit.commit_type
+            and commit.commit_type not in commit_types,
+            commits,
+        ),
+        key=lambda commit: commit.scope,
+    )
+
+
+def get_commit_trees(commits):
+
+    trees: List[CommitTree] = []
+    documentations = CommitTree(
+        commit_type=":notebook_with_decorative_cover: Documentation",
+        commits=get_commit_from_type(commits, "docs"),
+    )
+    if documentations.commits:
+        trees.append(documentations)
+    features = CommitTree(
+        commit_type=":rocket: Features", commits=get_commit_from_type(commits, "feat")
+    )
+    if features.commits:
+        trees.append(features)
+    fixes = CommitTree(
+        commit_type=":bug: Fixes", commits=get_commit_from_type(commits, "fix")
+    )
+    if fixes.commits:
+        trees.append(fixes)
+    reverts = CommitTree(
+        commit_type=":scream: Revert", commits=get_commit_from_type(commits, "revert")
+    )
+    if reverts.commits:
+        trees.append(reverts)
```

Comment:
Do I really need to read the five exact same `if`s ?

Reply:
right, but not my code 😢